wgpu/src/api/buffer.rs

```diff
@@ -775,20 +778,22 @@ impl AsRef<[u8]> for BufferView<'_> {
 ///
 /// [map]: Buffer#mapping-buffers
 #[derive(Debug)]
-pub struct BufferViewMut<'a> {
-    slice: BufferSlice<'a>,
+pub struct BufferViewMut {
+    buffer: Buffer,
```

Comment:
I suggest leaving a note indicating that this could be changed if `BufferSlice` ever changes:

```suggestion
    // `buffer, offset, size` are similar to `BufferSlice`, except that they own the buffer.
    buffer: Buffer,
```

Reply:
Done.

though: We could make BuffferSlice generic over Buffer (ref or owned), but that seems like overkill.